test.tsx

```diff
@@ -64,65 +65,77 @@ it('should give a react component', () => {
 
 it('should render', () => {
   const Foo2 = compile($injector)
-  const foo2 = renderIntoDocument(<Foo2 foo='hello' fooBar={42} />) as any
-  const element = $(findDOMNode(foo2))
-  expect(element.find('span').eq(0).text()).toBe('hello')
-  expect(element.find('span').eq(1).text()).toBe('504')
+  const element = document.createElement('div')
+  render(<Foo2 foo='hello' fooBar={42} />, element)
+  expect($(element).find('span').eq(1).text()).toBe('hello')
+  expect($(element).find('span').eq(2).text()).toBe('504')
+  unmountComponentAtNode(element)
 })
 
 it('should update', () => {
   const Foo2 = compile($injector)
   const element = document.createElement('div')
   render(<Foo2 foo='hello' fooBar={42} />, element)
-  expect($(element).find('span').eq(1).text()).toBe('504')
+  expect($(element).find('span').eq(2).text()).toBe('504')
+  unmountComponentAtNode(element)
   render(<Foo2 foo='hello' fooBar={43} />, element)
-  expect($(element).find('span').eq(1).text()).toBe('516')
+  expect($(element).find('span').eq(2).text()).toBe('516')
   unmountComponentAtNode(element)
 })
 
 it('should destroy', () => {
   const Foo2 = compile($injector)
   const element = document.createElement('div')
-  render(<Foo2 foo='hello' fooBar={42} />, element)
+  render(<Foo2 foo='hello' fooBar={42} />, element) as Element
   spyOn(FooController.prototype, '$onDestroy')
-  unmountComponentAtNode(element)
+  const unmounted = unmountComponentAtNode(element)
+  expect(unmounted).toBeTruthy
```

Comment:
Since we're doing more magic now, can you also assert that `element` is now empty?